iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/utils/executor/batch/utils/AlignedSeriesBatchCompactionUtils.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.storageengine.dataregion.compaction.execute.utils.executor.batch.utils;
+
+import org.apache.iotdb.db.storageengine.dataregion.compaction.execute.utils.executor.ModifiedStatus;
+
+import org.apache.tsfile.enums.TSDataType;
+import org.apache.tsfile.file.metadata.AlignedChunkMetadata;
+import org.apache.tsfile.file.metadata.IChunkMetadata;
+import org.apache.tsfile.read.TsFileSequenceReader;
+import org.apache.tsfile.read.common.TimeRange;
+import org.apache.tsfile.utils.Pair;
+import org.apache.tsfile.write.schema.IMeasurementSchema;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class AlignedSeriesBatchCompactionUtils {
+
+  private AlignedSeriesBatchCompactionUtils() {}
+
+  public static List<IMeasurementSchema> selectColumnBatchToCompact(
+      List<IMeasurementSchema> schemaList, Set<String> compactedMeasurements, int batchSize) {
+    List<IMeasurementSchema> selectedColumnBatch = new ArrayList<>(batchSize);
+    for (IMeasurementSchema schema : schemaList) {
+      if (!isLargeDataType(schema.getType())) {
+        continue;
+      }
+      if (compactedMeasurements.contains(schema.getMeasurementId())) {
+        continue;
+      }
+      compactedMeasurements.add(schema.getMeasurementId());
+      selectedColumnBatch.add(schema);
+      if (selectedColumnBatch.size() >= batchSize) {
+        return selectedColumnBatch;
+      }
+      if (compactedMeasurements.size() == schemaList.size()) {
+        return selectedColumnBatch;
+      }
+    }
+    for (IMeasurementSchema schema : schemaList) {
+      if (compactedMeasurements.contains(schema.getMeasurementId())) {
+        continue;
+      }
+      selectedColumnBatch.add(schema);
+      compactedMeasurements.add(schema.getMeasurementId());
+      if (selectedColumnBatch.size() >= batchSize) {
+        break;
+      }
+      if (compactedMeasurements.size() == schemaList.size()) {
+        break;
+      }
+    }
+    return selectedColumnBatch;
+  }
```

Comment:
We can not get chunk size before read ChunkHeader. It is hard to get chunk size of each series.

Reply:
Maybe we can record the maximum chunk size of each device or of the a file in TsFileResource?
It may be helpful to perform more strict memory control.